gematria/datasets/exegesis_benchmark.cc

```diff
@@ -45,19 +45,23 @@ static cl::opt<std::string> AnnotatedBlocksJson(
 
 Expected<BenchmarkCode> parseJSONBlock(
     const json::Object &BasicBlockJSON, MCInstPrinter &MachinePrinter,
-    const MCDisassembler &MachineDisassembler, const LLVMState &State) {
+    const MCDisassembler &MachineDisassembler, const LLVMState &State,
+    size_t BlockIndex) {
   BenchmarkCode BenchCode;
 
   std::optional<StringRef> HexValue = BasicBlockJSON.getString("Hex");
   if (!HexValue)
     return llvm::make_error<StringError>(
-        errc::invalid_argument, "Expected basic block to have hex value");
+        errc::invalid_argument, "Malformed basic block: Basic block at index " +
+                                    Twine(BlockIndex) + " has no hex value");
```

Comment:
The "Malformed basic block" part is repeated a couple of times. Given how lazy twines are, we can either pre-compute it - or move it into a lambda that can be called from here, so it doesn't need to be repeated.

Reply:
I added a `BasicBlockAtIndex` variable that holds that section and used it in all the errors.